libsolidity/codegen/CompilerUtils.cpp

```diff
@@ -36,11 +38,13 @@ namespace solidity
 
 const unsigned CompilerUtils::dataStartOffset = 4;
 const size_t CompilerUtils::freeMemoryPointer = 64;
+const size_t CompilerUtils::zeroPointer = CompilerUtils::freeMemoryPointer + 0x20;
 const unsigned CompilerUtils::identityContractAddress = 4;
 
 void CompilerUtils::initialiseFreeMemoryPointer()
 {
-	m_context << u256(freeMemoryPointer + 32);
+	solAssert(zeroPointer == CompilerUtils::freeMemoryPointer + 0x20, "");
```

Comment:
Can we just have a `static_assert` next to the const above?

Reply:
How would that help? I need to check that at this point here, the value of the free memory pointer does not overlap with the zero memory region.